File: src/main/kotlin/com/xs0/gqlktx/utils/StringOutputStream.java
package com.xs0.gqlktx.utils;

import org.jetbrains.annotations.NotNull;

import java.io.OutputStream;

public class StringOutputStream extends OutputStream {
    private final StringBuilder sb;

    public StringOutputStream() {
        sb = new StringBuilder();
    }

    public StringOutputStream(StringBuilder sb) {
        this.sb = sb;
    }

    public StringBuilder getStringBuilder() {
        return sb;
    }

    @Override
    public String toString() {
        return sb.toString();
    }

    @Override
    public void write(int b) {
        sb.append((char)(b & 255));
    }

    @Override
    public void write(@NotNull byte[] bytes) {
        for (byte b : bytes)
            sb.append((char)(b & 255));
    }

    @Override
    public void write(@NotNull byte[] b, int off, int len) {
        while (len-->0)
            sb.append((char)(b[off++] & 255));
    }
}

File: src/main/kotlin/com/xs0/gqlktx/utils/StringOutputStream.kt
package com.xs0.gqlktx.utils

import java.io.OutputStream

class StringOutputStream : OutputStream {
    val stringBuilder: StringBuilder

    constructor() {
        stringBuilder = StringBuilder()
    }

    constructor(sb: StringBuilder) {
        this.stringBuilder = sb
    }

    override fun toString(): String {
        return stringBuilder.toString()
    }

    override fun write(b: Int) {
        stringBuilder.append((b and 255).toChar())
    }

    override fun write(bytes: ByteArray) {
        for (b in bytes)
            stringBuilder.append((b.toInt() and 255).toChar())
    }

    override fun write(b: ByteArray, off: Int, len: Int) {
        var offset = off
        for (i in 1..len)
            stringBuilder.append((b[offset++].toInt() and 255).toChar())
    }
}
